app/domain/batch_process.rb

```diff
@@ -0,0 +1,17 @@
+class BatchProcess
+  attr_reader :output_documents
+
+  def initialize(output_documents)
+    @output_documents = output_documents
+  end
+
+  def csv
+    [headers].concat(output_documents.map(&:csv)).join
```

Comment:
With `header_row ` we still know that `output_documents` are rendered by `CSVRenderer`.